src/System.Net.Security/src/System/Net/CertificateValidationPal.Windows.cs

```diff
@@ -187,12 +186,12 @@ internal static X509Store EnsureStoreOpened(bool isMachineStore)
         {
             X509Store store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
 
-            if (store == null)
+            if (Volatile.Read(ref store) == null)
             {
                 lock (s_syncObject)
                 {
                     store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
-                    if (store == null)
+                    if (Volatile.Read(ref store) == null)
```

Comment:
I had that belief as well when writing the Unix version of this; but decided that it read as less confusing to treat it the same way both times; and the second volatile read should only ever happen once :).